### PR TITLE
fix: invalid ecosystem value in stack input

### PR DESCRIPTION
**Pull request by deepak1725**
Handle Invalid Ecosystem in Input Stack

Fixes: https://issues.redhat.com/browse/APPAI-1747

### CODE REVIEW

**Comment by prashbnair** [comment]
@deepak1725 the code fix looks fine, can you explain the reason for the Go statistics not being reported. Go is one of the supported ecosystems.

**Comment by deepak1725** [comment]
> @deepak1725 the code fix looks fine, can you explain the reason for the Go statistics not being reported. Go is one of the supported ecosystems.

I see there is unexpected `,` in ecosystem name(`golang,`) coming up as input which unable to match up already defined (Expected) dict keys of ecosystem values.
```
supported_eco = {"golang": 0, "npm": 0, "maven":0, "pypi": 0)
```
This fix cleans up the `ecosystem` value of any Non-Alpha Numeric values.

**Comment by prashbnair** [comment]
> I see there is unexpected , in ecosystem name(golang,) coming up as input which unable to match up already defined (Expected) dict keys of ecosystem values.

Is there something wrong in the way the ecosystem is getting stored for golang.  We should not be seeing golang,

**Comment by deepak1725** [comment]
> > I see there is unexpected , in ecosystem name(golang,) coming up as input which unable to match up already defined (Expected) dict keys of ecosystem values.
> 
> Is there something wrong in the way the ecosystem is getting stored for golang. We should not be seeing golang,

As far I analyse, there is definitely something unexpected stored in RDS via API Server.

File: f8a_report/v2/report_generator.py [hunk 1]
```diff
@@ -140,6 +142,11 @@ def analyse_stack(self, stacks_data, report_template) -> dict:
             stack = stack[0]
             stack_info_template = self.get_stack_info_template()
             ecosystem = stack.get('ecosystem')
+            if ecosystem not in self.supported_ecosystems:
+                logger.error("Unexpected Ecosystem type: %s, value: %s", type(ecosystem), ecosystem)
+                ecosystem = re.sub(r'\W+', '', ecosystem)
```
Comment (prashbnair) [inline f8a_report/v2/report_generator.py:147]: Why are you correcting the ecosystem and then checking again for error?

**Comment by prashbnair** [comment]
> As far I analyse, there is definitely something unexpected stored in RDS via API Server.

Can you please create another ticket and get that also fixed. Only then will the issue be resolved. Otherwise we will be missing out on some data in our reports.

File: f8a_report/v2/report_generator.py (hunk 1 above)
Comment (deepak1725) [reply f8a_report/v2/report_generator.py:147]: @prashbnair handling 2 cases here:
- when ecosystem= "golang,"-> Cleaning Ecosystem and then continuing process
- when ecosystem="golangg" or whatever -> Explicitly raising assert Error in First place to show it in logs

**Comment by deepak1725** [comment]
> > As far I analyse, there is definitely something unexpected stored in RDS via API Server.
> 
> Can you please create another ticket and get that also fixed. Only then will the issue be resolved. Otherwise we will be missing out on some data in our reports.

Here we go https://issues.redhat.com/browse/APPAI-1750

File: f8a_report/v2/report_generator.py (hunk 1 above)
Comment (prashbnair) [reply f8a_report/v2/report_generator.py:147]: I am not in favour of correcting and continuing, we will not know the errors. Let us  write the different ecosystems in the logs and continue. That way we can correct it.

Comment (deepak1725) [reply f8a_report/v2/report_generator.py:147]: Probably I was not clear enough
- when ecosystem= "golang,"-> Cleaning Ecosystem and then continuing process (Logging is there)
- when ecosystem="golangg" or whatever -> Cleaning is not possible, then Explicitly raising assert Error and show it in logs and stopping Process at very moment.

Comment (prashbnair) [reply f8a_report/v2/report_generator.py:147]: Lets us not fix anything here, why should we fix just one mistake, I feel it is not required and we need to fix the data corruption issue. Until then, let it continue to fail.

Comment (deepak1725) [reply f8a_report/v2/report_generator.py:147]: Ack. Closing PR

**Comment by deepak1725** [comment]
https://github.com/fabric8-analytics/f8a-stacks-report/pull/190#discussion_r577382531

**Pull request closed without merging**